odo/backends/sql.py

```diff
@@ -736,9 +742,15 @@ def select_or_selectable_to_frame(el, bind=None, dshape=None, **kwargs):
             if ty in datashape.integral:
                 dtypes[field] = 'float64'
             else:
-                dtypes[field] = ty.to_numpy_dtype()
+                try:
+                    dtypes[field] = ty.to_numpy_dtype()
+                except TypeError:
+                    dtypes[field] = np.dtype(object)
         else:
-            dtypes[field] = dtype.to_numpy_dtype()
+            try:
+                dtypes[field] = dtype.to_numpy_dtype()
+            except TypeError:
+                dtypes[field] = np.dtype(object)
```

Comment:
This is also required to support nullable foreign-keys as there's no numpy equivalent of a nullable it.

This whole block seems a bit redundant to me though - if the goal is to push it into pandas why bother going through numpy at all? pandas is much looser with types than numpy so it seems a bit pointless raising an exception on conversion to numpy when pandas could handle the data perfectly well (just using an object dtype)